x-pack/plugins/apm/public/context/UrlParamsContext.tsx

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { createContext, useReducer } from 'react';
+import { useLocation } from '../hooks/useLocation';
+import {
+  urlParamsReducer,
+  TimeRange,
+  refreshTimeRange,
+  resolveUrlParams,
+  IUrlParams
+} from '../store/urlParams';
+
+type RefreshTimeRangeFunction = (time: TimeRange) => void;
+
+const defaultRefresh = (() => null) as RefreshTimeRangeFunction;
+
+const UrlParamsContext = createContext({
+  urlParams: {} as IUrlParams,
+  refreshTimeRange: defaultRefresh
```

Comment:
nit: If you write the refresh handler like so, the type can be inferred. 

```suggestion
  refreshTimeRange: (time: TimeRange) => {}
```

I can see how one would prefer explicit types so your decision.